bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/TeaserImpl.java

```diff
@@ -522,14 +524,14 @@ public String getTitle() {
         @Nullable
         @Override
         public String getPath() {
-            Page page = (Page) ctaLink.getReference();
-            if (page != null) {
-                return page.getPath();
-            }
-            else {
-                // probably would make more sense to return null when not page is target, but we keep this for backward compatibility
-                return ctaLink.getURL();
+            if (ctaLink.getReference() instanceof Page) {
```

Comment:
`getCtaPage` can be reused

```
public String getPath() {
            return getCtaPage()
                    .map(Page::getPath)
                    .orElse(ctaLink.getURL());
        }
```

Reply:
Thx! Addressed in https://github.com/adobe/aem-core-wcm-components/pull/2343/commits/455757fb99d720cb69974f6235439db1c0dd0f62